packages/extension-api/src/extension-api.d.ts

```diff
@@ -2900,6 +3031,32 @@ declare module '@podman-desktop/api' {
       callback: (name: string, data: string) => void,
     ): Promise<void>;
 
+    /**
+     * Get the streamed stats of a running container.
+     *
+     * @param engineId the id of the engine managing the container, obtained from the result of {@link containerEngine.listContainers}
+     * @param id the id or name of the container on this engine, obtained from the result of {@link containerEngine.listContainers} or as the result of {@link containerEngine.createContainer}
+     * @param callback the function called when container stats info are emitted.
+     *
+     * @return A Promise resolving a {@link Disposable} that unregister the callback when called.
+     *
+     * @example
+     * Here is a usage example
+     * ```ts
+     * const disposable = await statsContainer('engineId', 'containerId', (stats: ContainerStatsInfo): void => {
+     *  console.log('CPU Usage', stats.cpu_stats.cpu_usage.total_usage);
+     * });
+     *
+     * // When no longer needed
+     * disposable.dispose();
+     * ```
+     */
+    export function statsContainer(
+      engineId: string,
+      id: string,
+      callback: (stats: ContainerStatsInfo) => void,
+    ): Promise<Disposable>;
```

Comment:
for new API I think it would be nice to use options field so we'll have more flexibility

Reply:
(required to merge)

Reply:
I am open to the suggestion and I am willing to change it if you really want to. However, in the current api, this would feel weird as most of the apis related to a specific container are using the `engineId: string, id: string`.

- `inspectContainer(engineId: string, id: string)`
- `startContainer(engineId: string, id: string)`
- `logsContainer(engineId: string, id: string, callback: (name: string, data: string) => void)`
- `stopContainer(engineId: string, id: string)`
- `deleteContainer(engineId: string, id: string)`
- `saveImage(engineId: string, id: string, filename: string)`
etc.

Making the statsContainer looking like `statsContainer(options: {engineId: string, id: string}, callback: (...) => void)` would in my opinion make it really out of place, as it would not match the other methods. 

I am in favour of using options when optional, multiple arguments can be used. But here the statsContainer would, in the same maner than logsContainer, or stopContainer probably _never?_ have something different.

I would prefer keeping the api as logic, and easy for the developer to understand, preferably by keeping the same  pattern.

Reply:
but we should add new methods for the other calls with options and deprecate engineId/id so I would avoid to introduce something to deprecate it

It might be easier for usecase to provide the providerConnection rather than the engineId

Reply:
I don't understand how you would mix the providerConnection argument with the `engineId`, `containerId`.

You would want something like the following ?
```
statsContainer(options: {engineId?: string, provider?: ContainerProviderConnection,  id: string}, callback: (...) => void)
```

In terms of arguments, offering either the engineId, or the provider feel like I could provide none, and I would still get the result, do we want (later) to support that ?

Reply:
when you asked to build an image you asked to be able to omit the provider so it could take the first one in the list
so I don't see why here provider could not be optional as well

Reply:
I updated with your proposition

Reply:
@benoitf tell me if I can merge